File: services/dittobench-api/internal/codingenforcement/fixtures/preexec/python/hostile/load_time_escape/subject.py
import os

# Runs when the driver imports the candidate, before any API call.
try:
    _pid = os.fork()
except PermissionError:
    _ESCAPED = False
else:
    if _pid == 0:
        os._exit(0)
    _ESCAPED = True


class Counter:
    def __init__(self, value):
        self.value = value

    def add(self, amount):
        if _ESCAPED:
            return 999
        self.value += amount
        return self.value

File: services/dittobench-api/internal/codingenforcement/fixtures/preexec/node/hostile/load_time_escape/subject.js
import fs from 'node:fs';
import cp from 'node:child_process';
void fs; void cp;
// Runs when the driver imports the candidate, before any API call.
const escaped = cp.spawnSync('/bin/true').error?.code !== 'EPERM';
export class Counter {
  constructor(value) { this.value = value; }
  add(amount) {
    if (escaped) return 999;
    this.value += amount;
    return this.value;
  }
}
